src/ray/common/client_connection.cc

```diff
@@ -242,7 +244,11 @@ void ClientConnection<T>::ProcessMessageHeader(const boost::system::error_code &
   }
 
   // If there was no error, make sure the protocol version matches.
-  RAY_CHECK(read_version_ == RayConfig::instance().ray_protocol_version());
+  if (!CheckProtocolVersion()) {
```

Comment:
It would be slightly clearer if we just have `CheckProtocolVersion()` here (not returning anything), and then inside of CheckProtocolVersion we call `ServerConnection<T>::Close();` where the comment says "and stop processing the connection".

Reply:
I think we need to skip the rest of this function if `CheckProtocolVersion` fails, thus it still needs to return a value.